physics/ephemeris_body.hpp

```diff
@@ -0,0 +1,318 @@
+        Vector<Acceleration, Frame>(
+            Instant const&)> intrinsic_acceleration,
+    Length const& length_integration_tolerance,
+    Speed const& speed_integration_tolerance,
+    AdaptiveStepSizeIntegrator<TimedBurnMotion> integrator,
+    Instant const& t) {}
+
+template<typename Frame>
+void Ephemeris<Frame>::AppendState(
+    typename NewtonianMotionEquation::SystemState const& state) {
+  last_state_ = state;
+  int index = 0;
+  for (auto& trajectory : oblate_trajectories_) {
+    trajectory->Append(
+        state.time.value,
+        DegreesOfFreedom<Frame>(state.positions[index].value,
+                                state.velocities[index].value));
+    ++index;
+  }
+  for (auto& trajectory : spherical_trajectories_) {
+    trajectory->Append(
+        state.time.value,
+        DegreesOfFreedom<Frame>(state.positions[index].value,
+                                state.velocities[index].value));
+    ++index;
+  }
+}
+
+template<typename Frame>
+template<bool body1_is_oblate,
+         bool body2_is_oblate>
+inline void Ephemeris<Frame>::ComputeOneBodyGravitationalAcceleration(
+    MassiveBody const& body1,
+    size_t const b1,
+    std::vector<not_null<MassiveBody const*>> const& bodies2,
+    size_t const b2_begin,
+    size_t const b2_end,
+    std::vector<Position<Frame>> const& positions,
+    not_null<std::vector<Vector<Acceleration, Frame>>*> const accelerations) {
+  GravitationalParameter const& μ1 =
+      body1.gravitational_parameter();
+  for (std::size_t b2 = std::max(b1 + 1, b2_begin); b2 < b2_end; ++b2) {
+    MassiveBody const& body2 = *bodies2[b2 - b2_begin];
+    GravitationalParameter const& μ2 =
+        body2.gravitational_parameter();
+
+    Displacement<Frame> const Δq = positions[b1] - positions[b2];
+
+    Exponentiation<Length, 2> const Δq_squared = InnerProduct(Δq, Δq);
+    // NOTE(phl): Don't try to compute one_over_Δq_squared here, it makes the
+    // non-oblate path slower.
+    Exponentiation<Length, -3> const one_over_Δq_cubed =
+        Sqrt(Δq_squared) / (Δq_squared * Δq_squared);
+
+    auto const μ1_over_Δq_cubed = μ1 * one_over_Δq_cubed;
+    (*accelerations)[b2] += Δq * μ1_over_Δq_cubed;
+
+    // Lex. III. Actioni contrariam semper & æqualem esse reactionem:
+    // sive corporum duorum actiones in se mutuo semper esse æquales &
+    // in partes contrarias dirigi.
+    auto const μ2_over_Δq_cubed = μ2 * one_over_Δq_cubed;
+    (*accelerations)[b1] -= Δq * μ2_over_Δq_cubed;
+
+    if (body1_is_oblate || body2_is_oblate) {
+      Exponentiation<Length, -2> const one_over_Δq_squared = 1 / Δq_squared;
+      if (body1_is_oblate) {
+        Vector<Acceleration, Frame> const order_2_zonal_acceleration1 =
+            Order2ZonalAcceleration<Frame>(
+                static_cast<OblateBody<Frame> const &>(body1),
+                Δq,
+                one_over_Δq_squared,
+                one_over_Δq_cubed);
+        (*accelerations)[b2] += order_2_zonal_acceleration1;
```

Comment:
This looks like it doesn't respect Newton's third law (see above)...
Investigate.
Later.